exporter/stackdriverexporter/stackdriver.go

```diff
@@ -152,30 +151,17 @@ func newStackdriverMetricsExporter(cfg *Config) (component.MetricsExporter, erro
 // pushMetrics calls StackdriverExporter.PushMetricsProto on each element of the given metrics
 func (me *metricsExporter) pushMetrics(ctx context.Context, m pdata.Metrics) (int, error) {
 	mds := pdatautil.MetricsToMetricsData(m)
-	dropped := 0
+	totalDropped := 0
 	for _, md := range mds {
-		d, err := me.mexporter.PushMetricsProto(ctx, md.Node, md.Resource, md.Metrics)
-		dropped += d
-
-		var points int
-		var st string
-		if err != nil {
-			points = d
-			s, ok := status.FromError(err)
-			if ok {
-				st = statusCodeToString(s)
-			}
-		} else {
-			_, points = pdatautil.TimeseriesAndPointCount(md)
-			st = "OK"
-		}
-		recordPointCount(ctx, points, st)
-
+		_, points := pdatautil.TimeseriesAndPointCount(md)
+		dropped, err := me.mexporter.PushMetricsProto(ctx, md.Node, md.Resource, md.Metrics)
+		recordPointCount(ctx, points-dropped, dropped, err)
+		totalDropped += dropped
 		if err != nil {
-			return dropped, err
+			return totalDropped, err
```

Comment:
Unrelated to your PR, but `totalDropped` number might be misleading here, if we have remaining unvisited metrics in `mds` which were also implicitly dropped by returning early, if I understand correctly.

Reply:
Yea this logic is a bit weird, and yes actually that seems like a bug that will lead to incorrectly reporting the number of dropped metrics.

Reply:
Changed to collect all errors and combine them instead of returning early on error.